sbi/neural_nets/ratio_estimators.py

```diff
@@ -0,0 +1,108 @@
+from abc import ABC, abstractmethod
+
+import torch
+from torch import Tensor, nn
+
+
+class RatioEstimator(nn.Module, ABC):
+    r"""Base class for ratio estimators.
+
+    The ratio estimator class is a wrapper around neural networks that enables
+    evaluation of `unnormalized_log_ratio` and the `loss` for `theta`, `x`
+    pairs. It also provides a method for combining (embedded) `theta` and `x`
+    into a single tensor.
+
+    Note:
+        We assume that the input to the ratio estimator is a tensor of shape
+        (sample, batch, event) where sample are iid draws, batch are
+        independent (not necessarily identically distributed) draws, and
+        event can be totally dependent.
+        See: https://bochang.me/blog/posts/pytorch-distributions/
+    """
+
+    def __init__(
```

Comment:
this signature will hold for all `RatioEstimators`?

Reply:
changed